core/sdk-core/src/main/java/software/amazon/awssdk/core/sync/RequestBody.java

```diff
@@ -49,11 +50,10 @@
     private final long contentLength;
     private final String contentType;
 
-    private RequestBody(ContentStreamProvider contentStreamProvider, long contentLength, String contentType) {
+    private RequestBody(ContentStreamProvider contentStreamProvider, Long contentLength, String contentType) {
         this.contentStreamProvider = paramNotNull(contentStreamProvider, "contentStreamProvider");
-        this.contentLength = contentLength;
+        this.contentLength = contentLength != null ? contentLength : Integer.MIN_VALUE;
```

Comment:
I think we should still validate it that it's not negative if not null so that it fails fast. 

Something like this:
https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/Validate.java#L622